torchrl/envs/common.py

```diff
@@ -801,6 +802,45 @@ def state_spec(self, value: CompositeSpec) -> None:
         finally:
             self.input_spec.lock_()
 
+    def step_and_maybe_reset(
+        self, tensordict: TensorDictBase
+    ) -> Tuple[TensorDictBase, TensorDictBase]:
+        """Runs a step in tne environment and possibly resets it.
+
+        This method returns the current tensordict updated with the `"next"`
+        value (as with ``step``) alongside the root tensordict of the next
+        step (ie, the result of ``step_mdp``).
+
+        This method will run :meth:`~.reset` on whichever sub-envs that needs
+        to be reset.
+
+        Returns: A ``cur_tensordict`` that will be used at the next step as the
+            root. It may contain data from a partial ``reset``. The second output
+            is the updated tensordict passed as input with the ``"next"`` key.
+
+        """
+        next_tensordict = self._step(tensordict)
+        next_tensordict = self._step_proc_data(next_tensordict)
+
+        done = next_tensordict.get(self.done_key)
+        truncated = next_tensordict.get("truncated", None)
+        if truncated is not None:
+            done = done | truncated
+
+        cur_td = _fuse_tensordicts(
```

Comment:
Why do we need this?

can't we just ` cur_td = self.reset(next_tensordict.clone())`